avocado/core/app.py

```diff
@@ -51,7 +54,13 @@ def __init__(self):
                 self.cli_cmd_dispatcher.map_method('configure', self.parser)
             if self.cli_dispatcher.extensions:
                 self.cli_dispatcher.map_method('configure', self.parser)
-            self.parser.finish()
+            try:
+                self.parser.finish()
+            except OptionValidationError as e:
+                log = logging.getLogger("avocado.app")
+                log.error(e)
+                STD_OUTPUT.close()
+                sys.exit(AVOCADO_JOB_FAIL)
```

Comment:
Oh I didn't notice this earlier. You don't need a special handling as we already have custom `error` inside our parser. You should raise `arparse.ArgumentError(self, msg)` in https://github.com/avocado-framework/avocado/pull/1324/commits/7a9f7ce4f694ffe385e276ed36550f36e636640e#diff-d54c4096c39dacc7b790b179120b796cR63 and it'll work properly without any addition here.